_includes/assets/js/model/utils.js

```diff
@@ -7,10 +7,12 @@
  * @param {Array} rows
  */
 function getUniqueValuesByProperty(prop, rows) {
-  return rows
-    .map(function(row) { return row[prop]; })
-    .filter(isElementUniqueInArray)
-    .filter(function(row) { return row; })
+  var uniques = new Set();
+  rows.forEach(function(r) {
+    uniques.add(r[prop])
+  });
+  return Array.from(uniques)
+    .filter(function(value) { return value; })
```

Comment:
what's the purpose of this filter?

Reply:
@leeoniya That is to filter out null values. I wonder if there is a better way, hopefully less mysterious?

Reply:
My this would be better:

```
var uniques = new Set();
rows.forEach(function(r) {
  if (r[prop]) {
    uniques.add(r[prop])
  }  
});
return Array.from(uniques).sort();
```

Reply:
oh, interesting. i didn't realize that the return value merely needs to *coerce* to `true`. i usually do an explit check to return a boolean. `.filter(v => v != null)` works for filtering just `null` and `undefined`. i guess that counts as less mysterious :D

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter#Description

Reply:
your `if` version works, too abd should compress better. again, i usually try to avoid coercion and do `if (r[prop] != null) {`.

you may want to bench thr `if` version against the `filter` version. if the vast majority of the values are non-null, running 92k `if`s may be slower than just plucking at the end. although i think the perf diff here will be negligible for datasets below millions of datapoints.

Reply:
Makes sense, thanks. I just committed that change. The array being filtered here is not the full list of rows - it's only the unique values from that Set - so I think going with the most readable code is fine here. Thanks for pointing that out!